core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -710,6 +710,24 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
       assert(runningTaskIds.isEmpty)
     }
   }
+
+  Seq(
+    ("local", 2, None),
+    ("local[2]", 3, None),
+    ("local[2, 1]", 3, None),
+    ("spark://test-spark-cluster", 2, Option(1)),
+    ("local-cluster[1, 1, 1000]", 2, Option(1)),
+    ("yarn", 2, Option(1))
+  ).foreach {
+    case (master, cpusPerTask, executorCores) =>
+      test(s"Avoid setting ${CPUS_PER_TASK.key} unreasonably when run on $master (SPARK-27192)") {
+        val conf = new SparkConf()
+        conf.set("master", master)
+        conf.set(CPUS_PER_TASK, cpusPerTask)
+        executorCores.map(executorCores => conf.set(EXECUTOR_CORES, executorCores))
+        intercept[SparkException] {sc = new SparkContext(conf)}
```

Comment:
Break the body onto a new line. Also, can you assert that the message contains a short substring that indicates the error is the expected one?